[MCH] add functionality and protections in mapping

pillot opened the pull request:
- add a function to loop over DE pad indices connected to a dual sampa.
- speedup the retrieval of cathode pad indices connected to a dual sampa.
- add protections in the conversions DsDetId <--> DsIndex

pillot committed:
add functionality and protections in MCH mapping

lmassacr commented:
Hi @pillot, 
I looked at the PR and it is good for me to approve once you have fixed the whitespace issue in Segmentation.h
One question thought, do you really want to keep both:
std::vector<int> getCatPadIndices(int dualSampaId) const;
and std::vector<int> catPadIndices(int dualSampaId) const;
as as far as I understood, they return the same information but one is just faster than the other? 
Cheers,
Laure

pillot commented:
Hi @lmassacr ,
catPadIndices(...) is the old function that I renamed and made private. It is used to fill the internal map of indices.
getCatPadIndices(...) is public and I changed it to return the indices stored in this map.
So there is only one visible function to get the indices, the fastest one, but both are needed.
Cheers,
Philippe

lmassacr reviewed (approved):
Hi @pillot, 
Thanks for the clarification. 
Cheers,
Laure
